src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SuddenHighwayTypeChangeCheck.java

```diff
@@ -0,0 +1,248 @@
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        if (TypePredicates.IS_EDGE.test(object) && ((Edge) object).isMainEdge()
+                && !isFlagged(object.getOsmIdentifier()))
+        {
+            final Edge edge = (Edge) object;
+            return HighwayTag.isCarNavigableHighway(edge) && HighwayTag.highwayTag(edge).isPresent()
+                    && edge.highwayTag().isMoreImportantThanOrEqualTo(this.minHighwayType)
+                    && !JunctionTag.isRoundabout(edge) && !JunctionTag.isCircular(edge);
+        }
+        return false;
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Edge edgeBeingVerified = (Edge) object;
+
+        final List<Edge> completeWayEdges = new ArrayList<>(
+                new OsmWayWalker(edgeBeingVerified).collectEdges());
+
+        final Node firstEdgeStartNode = completeWayEdges.get(0).start();
+        final Node lastEdgeEndNode = completeWayEdges.get(completeWayEdges.size() - 1).end();
+
+        final Set<Edge> firstEdgeStartNodeEdges = firstEdgeStartNode.connectedEdges();
+        firstEdgeStartNodeEdges
+                .removeIf(edge -> edge.getOsmIdentifier() == edgeBeingVerified.getOsmIdentifier());
+        final Set<Edge> lastEdgeEndNodeEdges = lastEdgeEndNode.connectedEdges();
+        lastEdgeEndNodeEdges
+                .removeIf(edge -> edge.getOsmIdentifier() == edgeBeingVerified.getOsmIdentifier());
+
+        HighwayTag edgeBeingVerifiedHighwayTag = HighwayTag.NO;
+        final Optional<HighwayTag> edgeBeingVerifiedHighwayTagOptional = HighwayTag
+                .highwayTag(edgeBeingVerified);
+        if (edgeBeingVerifiedHighwayTagOptional.isPresent())
+        {
+            edgeBeingVerifiedHighwayTag = edgeBeingVerifiedHighwayTagOptional.get();
+        }
+
+        final Set<HighwayTag> firstEdgeStartNodeEdgesHighwayTags = this
+                .getHighwayTags(firstEdgeStartNodeEdges);
+        final Set<HighwayTag> lastEdgeEndNodeEdgesHighwayTags = this
+                .getHighwayTags(lastEdgeEndNodeEdges);
+
+        // Check ways' first edge's connected edges for suspiciously large highway tag jumps.
+        for (final Edge firstEdgeEdge : firstEdgeStartNodeEdges)
+        {
+            final Optional<HighwayTag> firstEdgeEdgeHighwayTagOptional = HighwayTag
+                    .highwayTag(firstEdgeEdge);
+            if (firstEdgeEdgeHighwayTagOptional.isPresent()
+                    && edgeBeingVerifiedHighwayTagOptional.isPresent()
```

Comment:
this might need to be changed if you use the suggestion I made last ^
`&& !edgeBeingVerifiedHighwayTag.equals(HighwayTag.NO)`

Reply:
updated